### PR TITLE
tsdb: Replace sync/atomic with uber-go/atomic

### PR DESCRIPTION
**Pull request by jvrplmlmn**
Related with https://github.com/prometheus/prometheus/issues/7647, particularly for the `tsdb` package.

As this was my first contribution, I've preferred to keep the scope small and send additional PRs to keep iterating on https://github.com/prometheus/prometheus/issues/7647.

@codesome you are listed as a [maintainer](https://github.com/prometheus/prometheus/blob/master/MAINTAINERS.md) of this package, care to take a look? 👀

### CODE REVIEW

File: tsdb/db.go [hunk 1]
```diff
@@ -762,7 +762,7 @@ func (db *DB) Compact() (err error) {
 			break
 		}
 		mint := db.head.MinTime()
-		maxt := rangeForTimestamp(mint, db.head.chunkRange)
+		maxt := rangeForTimestamp(mint, db.head.chunkRange.Load())
```
Comment (johncming) [inline tsdb/db.go:765]: `chunkRange` only wrote on the init stage. Is it need to refactor?

Comment (bwplotka) [reply tsdb/db.go:765]: I think it's behind the lock, that's why it was not atomically changed/loaded so far. I think Load is fine for this though

File: tsdb/head.go [hunk 2]
```diff
@@ -53,11 +53,11 @@ var (
 type Head struct {
 	// Keep all 64bit atomically accessed variables at the top of this struct.
```
Comment (bwplotka) [inline tsdb/head.go:54]: We can remove all of those comments (:

File: tsdb/chunks/head_chunks.go [hunk 3]
```diff
@@ -80,7 +80,7 @@ func (e *CorruptionErr) Error() string {
 type ChunkDiskMapper struct {
 	// Keep all 64bit atomically accessed variables at the top of this struct.
 	// See https://golang.org/pkg/sync/atomic/#pkg-note-BUG for more info.
```
Comment (codesome) [inline tsdb/chunks/head_chunks.go:82]: We can remove this comment everywhere `go.uber.org/atomic` is used :)

Comment (codesome) [reply tsdb/chunks/head_chunks.go:82]: Ah, repeated comment :)

**Comment by jvrplmlmn** [comment]
@bwplotka @codesome comments addressed!

**Comment by codesome** [comment]
Thanks!

**Comment by codesome** [comment]
Realized after merging that we missed https://github.com/prometheus/prometheus/blob/master/tsdb/head_bench_test.go from tsdb.

**Pull request merged**